controllers/glanceapi_controller.go

```diff
@@ -807,9 +808,14 @@ func (r *GlanceAPIReconciler) reconcileNormal(ctx context.Context, instance *gla
 	}
 	// Mark the Deployment as Ready only if the number of Replicas is equals
 	// to the Deployed instances (ReadyCount), but mark it as True is Replicas
-	// is zero
-	if instance.Status.ReadyCount == *instance.Spec.Replicas || *instance.Spec.Replicas == 0 {
-		instance.Status.Conditions.MarkTrue(condition.DeploymentReadyCondition, condition.DeploymentReadyMessage)
+	// is zero. In addition, make sure the controller sees the last Generation
+	// by comparing it with the ObservedGeneration set in the StateFulSet.
+	if (instance.Status.ReadyCount == *instance.Spec.Replicas || *instance.Spec.Replicas == 0) &&
```

Comment:
side question: Why the Replicas == 0 case needs to be handled separately? I assume that in that case pthe ReadyCount  == Replicas condition would be true eventually as well. (definitely something for a separate PR)

Reply:
it was because `ReadyCount` wasn't always initialized, and if it doesn't exist that comparison will never be true. A different approach is to always initialize `ReadyCount = 0` and remove that additional condition. Definitely something I think it will be part of a follow up.

Reply:
https://github.com/openstack-k8s-operators/glance-operator/pull/502